dotcom-rendering/src/web/lib/useStickyVideo.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState, useRef } from 'react';
+import libDebounce from 'lodash.debounce';
+
+const useStickyVideo = (
+	options: IntersectionObserverInit & { debounce?: boolean },
+): [boolean, React.Dispatch<React.SetStateAction<HTMLElement | null>>] => {
+	const [isIntersecting, setIsIntersecting] = useState<boolean>(false);
+	const [node, setNode] = useState<HTMLElement | null>(null);
+
+	const observer = useRef<IntersectionObserver | null>(null);
+
+	// Enabling debouncing ensures the target element intersects for at least
+	// 200ms before the callback is executed
+	const intersectionFn: IntersectionObserverCallback = ([entry]) => {
+		if (entry.isIntersecting) {
+			setIsIntersecting(true);
+		} else {
+			setIsIntersecting(false);
+		}
+	};
+
+	const intersectionCallback = options.debounce
+		? libDebounce(intersectionFn, 200)
+		: intersectionFn;
+
+	useEffect((): any => {
+		if (observer.current) {
+			observer.current.disconnect();
+		}
+
+		if (window.IntersectionObserver) {
+			observer.current = new window.IntersectionObserver(
+				intersectionCallback,
+				options,
+			);
+
+			const { current: currentObserver } = observer;
+
+			if (node) {
+				currentObserver.observe(node);
+			}
```

Comment:
Is there a reason for destructuring, here?

Reply:
Just based on the pre-existing `useHasBeenSeen` hook, can change if required.

Reply:
Seems like calling `observer.current.observe(node)` would be more straightforward.

Reply:
Resolving as this hook has been deleted.